### ISSUE TITLE
feature request: add doc comments linting

### ISSUE THREAD

**Issue by lowlighter** [denoland/deno_lint#1279]
Since these are used on jsr.io for documentation (but even for `deno doc` or other doc generators), it'd be nice to add a few rules to lint comment docs like:

`no-undocumented-exported-member`: All exported members should have a doc comment:
```ts
export function bad() {}

/** This function does stuff. */
export function good() {}
```

`sentence-doc-comment`: All single line doc comment should starts with an uppercase letter and ends with a final dot.
```ts
/** this function does stuff */
export function bad() {}

/** This function does stuff. */
export function good() {}
```

`example-in-doc-comment`: All multiline doc comment should contain an example (maybe make this one optional ?)
```ts
/**
 * This function does stuff.
 * 
 *  Lorem ipsum... 
 *  Blah blah blah... 
 */
export function bad() {}

/**
 * This function does stuff.
 * 
 *  Lorem ipsum... 
 *  Blah blah blah...
 *
 * @example
 * ```ts
 * good() 
 * ```  
 */
export function good() {}
```
